index.js

```diff
@@ -1 +1,207 @@
+      }
+      return a.id - b.id;
+    });
+    response.render('allSightings', data);
+  });
+});
+
+// Route: Create form to input sighting
+app.get('/newSighting', (request, response) => {
+  read(FILENAME, (data, error) => {
+    if (error) {
+      response.sendStatus(500, error);
+      return;
+    }
+    response.render('submitNewSightingForm', data);
+  });
+});
+
+// Create a POST method / request to submit new sighting
+app.post('/newSighting', (request, response) => {
+  const requestBodyObj = convertToIterableObj(request.body);
+  // locally scoped boolean to track if current form is valid
+  let isThisFormValid = true;
+
+  // Perform input validation first
+  performValidationOnRequestBody(requestBodyObj, (feedbackMsg, areFieldsValid) => {
+    if (areFieldsValid === false) {
+      request.body.feedback = feedbackMsg;
+      isThisFormValid = false;
+      response.render('submitNewSightingForm', request.body);
+    }
+  });
+  if (isThisFormValid === false) {
+    return;
+  }
+
+  /**
+   * @param {FILENAME} contains all the data of the sightings to be read
+   * @param {key} the key to access in the file to be added
+   * @param {input} in this case, it is the iterated requestBodyObj
+   * @param {callback} performs a callback after taking in the data and error vars from add fn
+  */
+  add(FILENAME, 'sightings', requestBodyObj, (data, error) => {
+    // to be redirected to sighting/<index>
+    response.redirect(`/sighting/${data.sightings.length}`);
+  });
+});
+
+// Route: Render a particular sighting
+app.get('/sighting/:id', (request, response) => {
+  // id of sighting is not zero-indexed
+  const { id } = request.params;
+  read(FILENAME, (data) => {
+    const sightingData = data.sightings[id - 1];
+    response.render('sighting', sightingData);
+  });
+});
+
+// Render a form to edit a sighting
+app.get('/sighting/:id/edit', (request, response) => {
+  // id of sighting is not zero-indexed
+  const { id } = request.params;
+  read(FILENAME, (data) => {
+    const sightingData = data.sightings[id - 1];
+    response.render('editExistingSightingForm', sightingData);
+  });
+});
+
+// Edit the sighting through a PUT method
+app.put('/sighting/:id/edit', (request, response) => {
+  // id of sighting is not zero-indexed
+  const { id } = request.params;
+  const requestBodyObj = convertToIterableObj(request.body);
+
+  // locally scoped boolean to track if current form is valid
+  let isThisFormValid = true;
+
+  // Perform input validation first
+  performValidationOnRequestBody(requestBodyObj, (feedbackMsg, areFieldsValid) => {
+    if (areFieldsValid === false) {
+      // Assigning feedback & id from helper function into request.body
+      request.body.feedback = feedbackMsg;
+      // Note that request.body generated by all the forms does not include id automatically,
+      // hence the inclusion
+      request.body.id = id;
+      isThisFormValid = false;
+      response.render('sightingFormEdit', request.body);
```

Comment:
Is this a typo? Should it be `editExistingSightingForm`?

Reply:
ah yes, I think I changed it in my subsequent push